docs/system/charge_controller.md

```diff
@@ -1,9 +1,94 @@
+
+### Other types
+
+#### On/off regulators
+
+#### Shunt regulators
+
+### MPPT vs. PWM comparison
+
 ::: warning TODO
-- Why is a charge controller necessary and what does it do?
-- Different types of charge controllers
-    - PWM
-    - MPPT
-    - Others
+- Characteristic curve of solar and what is good about it
+- Interactive graph to explain different performance of MPPT and PWM
 :::
+
+The current-voltage and power-voltage curves of the panels are given in the below interactive graph as shown in Fig.1
+
+The red line indicates the battery voltage and the corrsponding power output when using a PWM-controller. Increasing the battery voltage to match the panels voltage would improve the systems performance but as soon as the panels voltage drops, no loading could be done anymore.
+
+<charge-controller-curve/>
+
+<figure>
+<center>
+    <figcaption><b>Figure 1.</b> Usable energy MPPT vs. PWM <b>(interactive)</b>.</figcaption>
+</center>
+</figure>
+
+#### Temperature influence
+
+Temperature has significant effect on the efficiency of charge controllers. As the temperature increases, $V_{oc}$ decreases i.e, current-voltage curve moves to the left but the current remains almost constant as seen from the interactive graph in Fig.1. Consequently, the power-voltage graph and hence maximum power point also moves towards lower value.
+Since the maximum power point moved to the lower value, the power transferred with a MPPT charge controller decreases. At this point, MPPT charge controllers can no longer outperform PWM charge controllers.
+
+## Wind
+
+For small wind turbines without variable-pitch speed-controller, a charge-controller can be used to keep the turbine at optimal rotation speed. Especially for wind speeds below the rated optimal wind, it is beneficial to reduce the load so that the turbine is not slowed down too much. For most turbines, the power coefficient $c_p$ is linked to the ratio between blade tips and the wind speed (TSR or $\lambda$) as shown in Figure [3]. A MPPT can be implemented in different ways. One approach could be to measure wind, rotational speed and have a pre-calculated look-up table for the power coefficient and regulate the accordingly. A major drawback is the pre-calculated look-up table, which is specific to one turbine and has to be verified with measurements in controlled conditions like a wind tunnel. It is also necessary to measure the wind which adds to the complexity. 
+
+Another, turbine-agnostic way is a dynamic approach where the duty cycle of the converter is permanently changed and the slope of the power production measured. Since the general curve is known (Figure [3]), this mehod will always approach the maximum point independently from any wind or rotational speed measurements. 
```

Comment:
What do you know by "since the general curve is known". Does it have to be measured?

Also: Can you state what the difference compared to solar MPPT is (if it exists)?